cloud/endagaweb/tasks.py

```diff
@@ -439,3 +439,20 @@ def req_bts_log(self, obj, retry_delay=60*10, max_retries=432):
         raise
     finally:
       obj.save()
+
+
+@app.task(bind=True)
+def unblock_blocked_subscribers(self):
+    """Unblock subscribers who are blocked for past 24 hrs.
+
+    This runs this as a periodic task managed by celerybeat.
+    """
+    unblock_time = django.utils.timezone.now() - datetime.timedelta(days=1)
+    subscribers = Subscriber.objects.filter(is_blocked=True,
+                                            block_time__lte=unblock_time)
+    if not subscribers:
+        return  # Do nothing
+    print 'Unblocking subscribers %s blocked for past 24 hours' % (
+        [subscriber.imsi for subscriber in subscribers], )
+    subscribers.update(is_blocked=False, block_time=None,
+                       block_reason='No reason to block yet!')
```

Comment:
we should also have another task to delete `SubscriberInvalidEvents` that are older than 24 hours? it seems like it would also need to update the `event_time` to the time of the 2nd invalid event in the negative actions list.